src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,56 @@
+package duke;
+
+import duke.command.*;
+
+public class Parser {
+    /**
+     * Interprets the user's input as a corresponding command.
+     *
+     * @param fullCommand The user's entire input.
+     * @return Command object that can be executed.
+     * @throws DukeException If the command cannot be recognised or data file cannot be accessed.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] spacedArr = fullCommand.split(" ", 2);
+        String command = spacedArr[0];
+        
+        if (command.equals("bye")) {
+            return new ExitCommand();
+        } else if (command.equals("list")) {
+            return new ListCommand();
+        }
+        
+        if (!command.equals("mark") && !command.equals("unmark") && !command.equals("todo") && !command.equals("deadline") && !command.equals("event") && !command.equals("delete") && !command.equals("find")) {
+            throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
```

Comment:
Instead of doing a check for unknown commands here, maybe it will be clearer if you shift this integrate this check into the switch statement below (L33-L55).

For example:

```java
switch (command) {
.
.
.
default:
    throw new DukeException("...");
}
```

This will remove the need to write long condition checks as the number of command type increase.

Reply:
I placed it on top in order to be before the "OOPS!!! The description cannot be empty." error, otherwise it would print that for any unknown commands of only one word :( I also couldn't shift the empty error to the end as I needed to ensure the commands have a description before the switch cases :(